plugin/builtin/helmchartinflationgenerator/HelmChartInflationGenerator_test.go

```diff
@@ -579,3 +580,137 @@ valuesInline:
 `)
 	th.AssertActualEqualsExpected(rm, "")
 }
+
+func TestHelmChartInflationGeneratorWithSameChartMultipleVersions(t *testing.T) {
+	th := kusttest_test.MakeEnhancedHarnessWithTmpRoot(t).
+		PrepBuiltin("HelmChartInflationGenerator")
+	defer th.Reset()
+	if err := th.ErrIfNoHelm(); err != nil {
+		t.Skip("skipping: " + err.Error())
+	}
+
+	tests := []struct {
+		name        string
+		chartName   string
+		repo        string
+		version     string
+		releaseName string
+	}{
+		{
+			name:        "terraform chart with no version grabs latest",
+			chartName:   "terraform",
+			repo:        "https://helm.releases.hashicorp.com",
+			version:     "",
+			releaseName: "terraform-latest",
+		},
+		{
+			name:        "terraform chart with version 1.1.1",
+			chartName:   "terraform",
+			repo:        "https://helm.releases.hashicorp.com",
+			version:     "1.1.1",
+			releaseName: "terraform-1.1.1",
+		},
+		{
+			name:        "terraform chart with version 1.1.1 again",
+			chartName:   "terraform",
+			repo:        "https://helm.releases.hashicorp.com",
+			version:     "1.1.1",
+			releaseName: "terraform-1.1.1-1",
+		},
+		{
+			name:        "terraform chart with version 1.1.2",
+			chartName:   "terraform",
+			repo:        "https://helm.releases.hashicorp.com",
+			version:     "1.1.2",
+			releaseName: "terraform-1.1.2",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			config := fmt.Sprintf(`
+apiVersion: builtin
+kind: HelmChartInflationGenerator
+metadata:
+ name: %s
+name: %s
+version: %s
+repo: %s
+releaseName: %s
+`, tt.chartName, tt.chartName, tt.version, tt.repo, tt.releaseName)
+
+			rm := th.LoadAndRunGenerator(config)
+			assert.True(t, len(rm.Resources()) > 0)
+
+			chartDir := fmt.Sprintf("charts/%s", tt.chartName)
+			if tt.version != "" {
+				chartDir = fmt.Sprintf("charts/%s-%s/%s", tt.chartName, tt.version, tt.chartName)
```

Comment:
nit: Similar to above, let's convert lines 645-648 to an if/else for readability

Reply:
https://github.com/kubernetes-sigs/kustomize/pull/4999/commits/380abb9e79ace1b03d15985e885e4b9e42c78ac7